src/IntroPage/index.js

```diff
@@ -119,22 +118,20 @@ const Intro = (props: Props) => (
     </div>
     <div className="buttons">
       <div className="primary">
-        <Link to={routeDefinitions.SIGN_IN}>
-          <button
-            onClick={() =>
-              getSessionToken()
-                ? props.history.push(routeDefinitions.UPCOMING_BOOKING)
-                : props.history.push(routeDefinitions.SIGN_IN)
-            }
-          >
-            I have an existing booking
-          </button>
-        </Link>
+        <button
+          onClick={() =>
```

Comment:
this should be probably defined as named function to avoid recreating new anonymous function on every render